storage/innobase/mtr/mtr0mtr.cc

```diff
@@ -1042,13 +1042,19 @@ std::pair<lsn_t,byte*> log_t::append_prepare(size_t size, bool ex) noexcept
   size_t b{spin ? lock_lsn() : buf_free.load(std::memory_order_relaxed)};
   write_to_buf++;
 
-  const lsn_t l{lsn.load(std::memory_order_relaxed)}, end_lsn{l + size};
+  lsn_t l{lsn.load(std::memory_order_relaxed)}, end_lsn{l + size};
 
   if (UNIV_UNLIKELY(pmem
                     ? (end_lsn -
                        get_flushed_lsn(std::memory_order_relaxed)) > capacity()
                     : b + size >= buf_size))
+  {
     b= append_prepare_wait<spin>(b, ex, l);
+    /* While flushing log, we had released the log system latch and LSN could
+    have progressed in the meantime. */
+    l = lsn.load(std::memory_order_relaxed);
+    end_lsn = l + size;
```

Comment:
I think that the comment is slightly misleading, speaking about "the log system latch" while there are multiple latches in `log_sys`. An intuitive interpretation of the term would be `log_sys.latch`, but that is wrong. (Holding an exclusive `log_sys.latch` would freeze `log_sys.lsn` but most writers are holding a shared `log_sys.latch`.) What is relevant is `lsn_lock` or `lock_lsn()`. They protect the allocation of LSN. So, the comment should be reworded.

Furthermore, the [CODING_STANDARDS.md](https://github.com/MariaDB/server/blob/11.6/CODING_STANDARDS.md) linked from the pull request description says that there should be no space before an assignment operator. I would suggest the following:
```c++
    /* The LSN could change while we were not holding lsn_lock or lock_lsn() */
    l= lsn.load(std::memory_order_relaxed);
    end_lsn= l + size;
```

Reply:
right, it is the lsn_lock. Thanks for pointing it out. And I keep forgetting about the assignment rule :)